client/allocrunner/taskrunner/validate_hook.go

```diff
@@ -43,7 +43,7 @@ func validateTask(task *structs.Task, taskEnv *taskenv.TaskEnv, conf *config.Con
 	var mErr multierror.Error
 
 	// Validate the user
-	unallowedUsers := conf.ReadStringListToMapDefault("user.blacklist", config.DefaultUserBlacklist)
+	unallowedUsers := conf.ReadStringListToMapDefault("user.denylist", config.DefaultUserDenylist)
```

Comment:
We'll want to update `ReadStringListToMapDefault` to do the same sort of thing we've done with `ReadStringListToMap` so that we can keep backwards compatibility with `user.blacklist`.